### PR TITLE
Exporting android login spec

### PR DESCRIPTION
**Pull request by tparwani1**

### CODE REVIEW

File: spec/support/page_objects/android/screens/choose_account.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+
+module Android
+  module Screens
+    class ChooseAccount < Base
+      self.page_name = 'ChooseAccount'
+
+      element :choose_account, :xpath, '/hierarchy/android.widget.FrameLayout/android.widget.LinearLayout/android.widget.FrameLayout/android.view.ViewGroup/android.widget.FrameLayout[1]/android.view.ViewGroup/android.widget.TextView'
```
Comment (ineverov) [inline spec/support/page_objects/android/screens/choose_account.rb]: is there any other locator? should we ask dev to add it?

File: spec/support/page_objects/android/screens/device_challenge.rb [hunk 2]
```diff
@@ -5,10 +5,10 @@ module Screens
     class DeviceChallenge < Base
       self.page_name = 'DeviceChallenge'
 
-      elements :texts, :xpath, '//XCUIElementTypeStaticText'
-
-      element :answer_field,    :accessibility_id, 'challengeAnswerTextField'
-      element :continue_button, :accessibility_id, 'Continue'
+      # elements :texts, :xpath, '//XCUIElementTypeStaticText'
+      element :challenge_question_text, :accessibility_id, 'deviceChallengeEditText'
```
Comment (ineverov) [inline spec/support/page_objects/android/screens/device_challenge.rb]: looks like it is not used

File: spec/support/page_objects/android/screens/choose_account.rb (hunk 1 above)
Comment (tparwani1) [reply spec/support/page_objects/android/screens/choose_account.rb]: yes i have asked devs for the id, they should include next time they do resin/refactor

File: spec/features/login_spec.rb [hunk 3]
```diff
@@ -56,6 +56,16 @@
       account_summary_screen.wait_until_loaded
     end
 
+    it 'Remeber me switch', disco: :happy_path do
+      login_screen.login(username: 'appledemouser', password: 'aa')
+      if app.challenge_question?
+        challenge_screen.wait_until_loaded
+        challenge_screen.answer_correct_challenge_question
+      end
+      login_screen.hit_back_button(4)
```
Comment (ineverov) [inline spec/features/login_spec.rb]: should we have keycode inside the method?

File: spec/support/page_objects/android/screens/device_challenge.rb (hunk 2 above)
Comment (tparwani1) [reply spec/support/page_objects/android/screens/device_challenge.rb]: its now being used for MFA code

File: etc/config.yml.example [hunk 4]
```diff
@@ -4,6 +4,7 @@ providers:
   local:
     android:
       caps:
+        app: /Users/tparwani/projects/android/app/build/outputs/apk/staging/debug/OneMain-v7.16.0-dev-staging-debug.apk
```
Comment (himankarbn) [inline etc/config.yml.example]: can you use variable like one you did in android_automation? Though we are planning to use url to get the file but for now lets make this as a variable

Comment (ineverov) [reply etc/config.yml.example]: @himankarbn It's just an example config. You need to copy it to etc/local/config.yml and modify with your values

@smelus ^^ FYI as well

File: spec/support/page_objects/android/screens/choose_account.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Android
+  module Screens
+    class ChooseAccount < Base
+      self.page_name = 'ChooseAccount'
+
+      element :choose_account, :xpath, '/hierarchy/android.widget.FrameLayout/android.widget.LinearLayout/android.widget.FrameLayout/android.view.ViewGroup/android.widget.FrameLayout[1]/android.view.ViewGroup/android.widget.TextView'
```
Comment (ineverov) [inline spec/support/page_objects/android/screens/choose_account.rb]: Is there any more precise/stable locator?

File: spec/support/page_objects/android/screens/device_challenge.rb [hunk 6]
```diff
@@ -7,31 +7,37 @@ module Screens
...
-        texts.last.value.strip
+        Waiter.wait_for(timeout: 5) { answer_field.text }.split.last.delete('?')
       end
 
       def answer_correct_challenge_question
         enter_challenge_answer(device_answer)
         sleep 2
       end
+
+      def enter_MFA_code(code = nil)
```
Comment (ineverov) [inline spec/support/page_objects/android/screens/device_challenge.rb]: prefer snail_case for method names, i.e `enter_mfa_code`

File: spec/support/page_objects/android/screens/forget_username.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        first_name_field.set first_name
+        last_name_field.set last_name
+        account_number_field.set account_number
+        ssn_last4_field.set ssn_last4
+        zip_field.set zip
+      end
+
+      def flash_error(message:)
+        flash_error_message.text.should == message
+         end
```
Comment (ineverov) [inline spec/support/page_objects/android/screens/forget_username.rb]: align pls

File: spec/support/page_objects/android/screens/login.rb [hunk 8]
```diff
@@ -15,12 +15,18 @@ class Login < ::Android::Screens::Base
...
 
       element :forgot_username, :accessibility_id, 'forgotUsernameButton'
       element :forgot_password, :accessibility_id, 'forgotPasswordButton'
 
       element :flash_error_message, :accessibility_id, 'alertMessageTextView'
 
+      element :legal_and_privacy, :accessibility_id, 'legalAndPrivacyButton'
+      element :disclosures, :accessibility_id, 'disclosuresButton'
+
+      element :URL, :id, 'com.android.browser:id/url'
```
Comment (ineverov) [inline spec/support/page_objects/android/screens/login.rb]: snail case pls `url`

File: spec/support/page_objects/common/screens/base.rb [hunk 9]
```diff
@@ -57,6 +57,10 @@ def scroll_and_wait(**args)
       def alert(**args)
         Capybara.page.driver.appium_driver.alert(**args)
       end
+
+      def hit_back_button(args)
```
Comment (ineverov) [inline spec/support/page_objects/common/screens/base.rb]: why do you want it to accept args? should we name it `hit_back_button()` with `Capybara.page.driver.appium_driver.press_keycode(4)` and call it from code as
`login_screen.hit_back_button`

File: spec/support/page_objects/ios/screens/device_challenge.rb [hunk 10]
```diff
@@ -32,6 +35,15 @@ def answer_correct_challenge_question
         enter_challenge_answer(device_answer)
         sleep 2
       end
+
+      def enter_MFA_code(code = nil)
```
Comment (ineverov) [inline spec/support/page_objects/ios/screens/device_challenge.rb]: `enter_mfa_code`

File: spec/support/page_objects/ios/screens/login.rb [hunk 11]
```diff
@@ -8,16 +8,28 @@ module Screens
...
+      element :login_button, :accessibility_id, 'LOG IN'
+      element :logout_button, :accessibility_id, 'Yes, Log Out'
 
       elements :email_errors,    :accessibility_id, 'emailErrorLabel'
       elements :password_errors, :accessibility_id, 'passwordErrorLabel'
 
+      element :legal_and_privacy, :accessibility_id, 'Legal & Privacy'
+      element :disclosures, :accessibility_id, 'Disclosures'
+
+      element :URL_done_button, :accessibility_id, 'Done'
```
Comment (ineverov) [inline spec/support/page_objects/ios/screens/login.rb]: `url_done_button` and `url`

File: spec/features/login_spec.rb [hunk 12]
```diff
@@ -5,135 +5,179 @@
...
       login_screen.login(username: '', password: '')
       expect(login_screen.email_field_error.strip).to eql 'This field is required.'
       login_screen.login(username: 'aa', password: '')
       expect(login_screen.password_field_error.strip).to eql 'This field is required.'
     end
 
-    it 'multiple email address found' do
-      login_screen.login(username: 'dgotech@iloan.com', password: 'Passw0rd')
+    it 'Inactive Account' do
+      login_screen.login(username: 'docrtvtest99', password: 'New1User')
```
[the remark this answers is not in the document]
Comment (ineverov) [reply spec/features/login_spec.rb]: 👍

File: spec/support/page_objects/common/screens/base.rb [hunk 13]
```diff
@@ -57,6 +57,10 @@ def scroll_and_wait(**args)
       def alert(**args)
         Capybara.page.driver.appium_driver.alert(**args)
       end
+
+      def hit_back_button()
```
Comment (ineverov) [inline spec/support/page_objects/common/screens/base.rb]: `def hit_back_button` I put parentheses to show there are no parameters. Rubocop will complain about `()`

File: spec/features/login_spec.rb [hunk 14]
```diff
@@ -5,135 +5,149 @@
...
 
-    xit 'customer with no challenge questions' do
-      login_screen.fill_login_form(username: 'no_challenge_questions', password: 'password')
-      login_screen.login_button.click
-      account_summary_screen.wait_until_loaded
+    it 'Disclosure Page' do
+      login_screen.disclosures.click
+      if app.andriod?
+        expect(login_screen.url.text).to include '/disclosures'
+        login_screen.hit_back_button()
```
Comment (ineverov) [inline spec/features/login_spec.rb]: remove `()`

File: spec/features/login_spec.rb [hunk 15]
```diff
@@ -5,135 +5,149 @@
...
-      login('disabled', 'Passw0rd')
-      expect(error_message).to have_content 'Your account is disabled'
+    it 'Remeber me switch', disco: :happy_path do
+      login_screen.login(username: 'appledemouser', password: 'aa')
+      challenge_screen.wait_until_loaded
+      challenge_screen.answer_correct_challenge_question
+      if app.ios?
+          choose_account_screen.select_account
+          account_summary_screen.logout.click
+          login_screen.remember_me_switch_check
```
Comment (ineverov) [inline spec/features/login_spec.rb]: What does this method do?
I'd rename `logout` to `logout_button` in the Base page. And method `logout` to click on it.
and check to be in the test itself.
Something like this.
```
if app.ios?
  choose_account_screen.select_account
  account_summary_screen.logout 
elsif app.android?
  account_summary_screen.hit_back_button
else
  raise 'not supported platform'
end

expect(login_screen.username_field.text).to eql('appledemouser')
```

File: spec/features/login_spec.rb [hunk 16]
```diff
@@ -5,135 +5,149 @@
...
     end
 
-    xit 'customer with no challenge questions' do
-      login_screen.fill_login_form(username: 'no_challenge_questions', password: 'password')
-      login_screen.login_button.click
-      account_summary_screen.wait_until_loaded
+    it 'Disclosure Page' do
+      login_screen.disclosures.click
+      if app.andriod?
+        expect(login_screen.url.text).to include '/disclosures'
```
Comment (ineverov) [inline spec/features/login_spec.rb]:
```
# Ios::Screens::Disclosures

load_validation do
  [url.text == '/disclosures', 'Failed to load disclosures page. Current url #{url.text}']
end
``` 
`app.disclosures.wait_until_loaded`

File: spec/support/page_objects/ios/screens/forget_password.rb [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+      element :flash_error_message, :accessibility_id, 'bannerMessage'
+      element :username_error, :accessibility_id, 'usernameErrorLabel'
+
+      def enter_username(username:)
+        email_or_username.set username
+        continue_buton.click
+      end
+
+      def flash_error(message:)
+        flash_error_message.text.should == message
```
Comment (ineverov) [inline spec/support/page_objects/ios/screens/forget_password.rb]: Please, no assertions in page objects

**Pull request merged**